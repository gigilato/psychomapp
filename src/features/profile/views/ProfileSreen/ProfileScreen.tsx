import { router } from 'expo-router'

import { Box, KeyboardAvoidingView, Separator, Text } from '$atoms'
import { Header } from '$features/navigation/views'
import { signOutAsync, useAuthStore } from '$infra/auth'
import { i18n } from '$infra/i18n'

import { ProfileSetting } from './components'

export const ProfileScreen = () => {
  const { firstname, lastname, job, gender, email } = useAuthStore((state) => state.profile!)
  return (
    <KeyboardAvoidingView>
      <Header className="items-center justify-center">
        <Text variant="title" className="text-white-light">
          {`${firstname} ${lastname}`}
        </Text>
        <Text variant="subtitle" className="text-white-light" textTransform="upper-first">
          {i18n.t(`profile.jobs.${job}` as any, { context: gender })}
        </Text>
      </Header>
      <Box className="px-l">
        <ProfileSetting ID="profile/email" content={email} icon="envelope" />
        <ProfileSetting
          ID="profile/password"
          content={i18n.t('settings.updatePassword')}
          icon="lock"
          type="link"
        />
      </Box>
      <Separator className="my-l" />
      <Box className="px-l">
        <ProfileSetting ID="profile/test" content={i18n.t('settings.tests')} type="navigation" />
        <ProfileSetting
          ID="profile/equipment"
          content={i18n.t('settings.equipment')}
          type="navigation"
          onPress={() => router.push('/dashboard/profile/equipment/')}
        />
        <ProfileSetting
          ID="profile/billing"
          content={i18n.t('settings.billing')}
          type="navigation"
        />
      </Box>
      <Separator className="my-l" />
      <ProfileSetting
        ID="profile/signOut"
        content={i18n.t('settings.cgu')}
        type="link"
        className="px-l"
      />
      <Separator className="my-l" />
      <ProfileSetting
        ID="profile/signOut"
        content={i18n.t('common.signOut')}
        icon="log-out"
        color="darkRed"
        onPress={signOutAsync}
        className="px-l"
      />
    </KeyboardAvoidingView>
  )
}
