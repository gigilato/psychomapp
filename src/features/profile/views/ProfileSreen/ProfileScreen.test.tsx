import { render } from '@testing-library/react-native'
import React from 'react'

import { ProfileScreen } from './ProfileScreen'

describe('TabBar', () => {
  it('should renders correctly', () => {
    const { toJSON } = render(<ProfileScreen />)
    expect(toJSON()).toMatchSnapshot()
  })
})
