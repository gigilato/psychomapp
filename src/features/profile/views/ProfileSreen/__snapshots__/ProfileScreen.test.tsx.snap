// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBar should renders correctly 1`] = `
[
  <RCTScrollView
    contentContainerStyle={
      [
        {
          "flexGrow": 1,
        },
        undefined,
      ]
    }
    onScroll={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={16}
    style={
      [
        [
          {
            "transform": [
              {
                "translateY": 0,
              },
            ],
          },
          {
            "backgroundColor": "#FFFCF9",
          },
          undefined,
        ],
      ]
    }
  >
    <View>
      <View
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            [
              undefined,
              {
                "height": 200,
                "width": 750,
              },
            ],
          ]
        }
      >
        <View
          style={
            {
              "height": 200,
              "width": 750,
            }
          }
        >
          <View
            color="#BB99DD"
            path={
              {
                "addCircle": [MockFunction],
                "close": [MockFunction] {
                  "calls": [
                    [],
                  ],
                  "results": [
                    {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
                "cubicTo": [MockFunction] {
                  "calls": [
                    [
                      187.5,
                      180,
                      375,
                      200,
                      750,
                      190,
                    ],
                  ],
                  "results": [
                    {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
                "lineTo": [MockFunction] {
                  "calls": [
                    [
                      0,
                      150,
                    ],
                    [
                      750,
                      0,
                    ],
                  ],
                  "results": [
                    {
                      "type": "return",
                      "value": undefined,
                    },
                    {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
                "moveTo": [MockFunction] {
                  "calls": [
                    [
                      0,
                      0,
                    ],
                  ],
                  "results": [
                    {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
              }
            }
            strokeWidth={2}
            style="fill"
          />
        </View>
        <View
          style={
            [
              {
                "alignItems": "center",
              },
              {
                "justifyContent": "center",
              },
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Bold",
                },
                {
                  "fontSize": 34,
                },
                {
                  "lineHeight": 42,
                },
                {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            firstname lastname
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Bold",
                },
                {
                  "fontSize": 20,
                },
                {
                  "lineHeight": 28,
                },
                {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Profile.jobs.job
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "paddingLeft": 24,
              "paddingRight": 24,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          animation="none"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "alignItems": "center",
              },
              {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              [
                {},
                undefined,
              ],
            ]
          }
          testID="profile/email"
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#FFFACD",
                },
                {
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                },
                {
                  "marginRight": 12,
                },
                {
                  "borderBottomLeftRadius": 6,
                  "borderBottomRightRadius": 6,
                  "borderTopLeftRadius": 6,
                  "borderTopRightRadius": 6,
                },
                [
                  undefined,
                  {
                    "backgroundColor": "#f5f7fa",
                  },
                ],
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            email@email.com
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          animation="opacity"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "alignItems": "center",
              },
              {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              [
                {
                  "opacity": undefined,
                },
                undefined,
              ],
            ]
          }
          testID="profile/password"
        >
          <View
            style={
              [
                {
                  "backgroundColor": "#FFFACD",
                },
                {
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                },
                {
                  "marginRight": 12,
                },
                {
                  "borderBottomLeftRadius": 6,
                  "borderBottomRightRadius": 6,
                  "borderTopLeftRadius": 6,
                  "borderTopRightRadius": 6,
                },
                [
                  undefined,
                  {
                    "backgroundColor": "#f5f7fa",
                  },
                ],
              ]
            }
          >
            <Text />
          </View>
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            Changer le mot de passe
          </Text>
          <Text />
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#D3D3D3",
            },
            {
              "marginBottom": 24,
              "marginTop": 24,
            },
            [
              undefined,
              [
                {
                  "height": 0.5,
                },
                undefined,
              ],
            ],
          ]
        }
      />
      <View
        style={
          [
            {
              "paddingLeft": 24,
              "paddingRight": 24,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          animation="opacity"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "alignItems": "center",
              },
              {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              [
                {
                  "opacity": undefined,
                },
                undefined,
              ],
            ]
          }
          testID="profile/test"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            Mon classeur de tests
          </Text>
          <Text />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          animation="opacity"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "alignItems": "center",
              },
              {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              [
                {
                  "opacity": undefined,
                },
                undefined,
              ],
            ]
          }
          testID="profile/equipment"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            Mon matériel
          </Text>
          <Text />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          animation="opacity"
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "flexDirection": "row",
              },
              {
                "alignItems": "center",
              },
              {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              [
                {
                  "opacity": undefined,
                },
                undefined,
              ],
            ]
          }
          testID="profile/billing"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "SpaceGrotesk-Regular",
                },
                {
                  "fontSize": 16,
                },
                {
                  "lineHeight": 18,
                },
                {
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            Ma facturation
          </Text>
          <Text />
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#D3D3D3",
            },
            {
              "marginBottom": 24,
              "marginTop": 24,
            },
            [
              undefined,
              [
                {
                  "height": 0.5,
                },
                undefined,
              ],
            ],
          ]
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        animation="opacity"
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 12,
              "paddingTop": 12,
            },
            {
              "paddingLeft": 24,
              "paddingRight": 24,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID="profile/signOut"
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          Conditions générales d'utilisation
        </Text>
        <Text />
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "#D3D3D3",
            },
            {
              "marginBottom": 24,
              "marginTop": 24,
            },
            [
              undefined,
              [
                {
                  "height": 0.5,
                },
                undefined,
              ],
            ],
          ]
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        animation="none"
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "alignItems": "center",
            },
            {
              "paddingBottom": 12,
              "paddingTop": 12,
            },
            {
              "paddingLeft": 24,
              "paddingRight": 24,
            },
            [
              {},
              undefined,
            ],
          ]
        }
        testID="profile/signOut"
      >
        <View
          style={
            [
              {
                "backgroundColor": "#FFFACD",
              },
              {
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              },
              {
                "marginRight": 12,
              },
              {
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
              },
              [
                undefined,
                {
                  "backgroundColor": "#F4E9E9",
                },
              ],
            ]
          }
        >
          <Text />
        </View>
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              },
            ]
          }
        >
          Déconnexion
        </Text>
      </View>
    </View>
  </RCTScrollView>,
  <View
    style={
      [
        {
          "backgroundColor": "#BB99DD",
        },
        {
          "position": "absolute",
        },
        {
          "top": 0,
        },
        {
          "left": 0,
        },
        {
          "right": 0,
        },
        [
          undefined,
          {
            "height": 0,
          },
        ],
      ]
    }
  />,
]
`;
